### PR TITLE
Add example for detecting and troubleshooting audio issues

### PR DESCRIPTION
**Pull request by anhallbe**

### CODE REVIEW

**Comment by stefanalund** [comment]
Please add to list of examples in `index.html` as well as `README.md`.

File: audioDetector.html [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+ * Only one DeviceSource can be monitored at any given time.
+ * 
+ * Usage:
+ * var audioMonitor = new AudioMonitor()
+ * audioMonitor.on('event_type', callback)
+ * 
+ * audioMonitor.monitor(someDeviceSourceWithAudio)
+ */
+function AudioMonitor () {
+  cct._EmitterMixin.call(this)
```
Comment (Rugvip) [inline audioDetector.html:84]: Should handle this by passing in some callbacks instead imo, since it's private API.

File: audioDetector.html [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+  // Called when DeviceSource.promise is rejected, for instance if access is denied or because of some other error.
+  _handleDeviceRejected: function (error) {
+    this._emit('error', error)
+    if (error.name === 'PermissionDeniedError' || error.name === 'SecurityError' || error.name === 'NotAllowedError') {
+      // Browsers are inconsistent in what errors are thrown when mic/camera access is denied.
+      // PermissionDeniedError is thrown by Chrome and the plugin
+      // SecurityError is from Firefox 48
+      // NotAllowedError is from Firefox 50 Developer edition and onward
+      this._emit('PermissionDeniedError', error)
```
Comment (Rugvip) [inline audioDetector.html:126]: The spec says the error should be called 'NotAllowedError', so let's use that instead.

Comment (Rugvip) [inline audioDetector.html:126]: Actually the above should be handled within the library, will fix.

**Comment by Rugvip** [comment]
Merging this now, let's update it once the error handling stuff is released in libcct.

**Pull request merged**